### PR TITLE
HX711 amplifier support

### PR DESCRIPTION
**Pull request by lukfud**

### CODE REVIEW

File: src/supla/sensor/HX711.h [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  }
+
+  float getCalFactor() {
+    return calFactor;
+  }
+
+ protected:
+  ::HX711_ADC hx711Sensor;
+  int sdaPin;
+  int sclPin;
```
Comment (klew) [inline src/supla/sensor/HX711.h:149]: those variables are not used. Remove it

File: src/supla/sensor/HX711.h [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  float value = 0;
+  uint32_t tareOffset = 0;
+  float calFactor;
```
Comment (klew) [inline src/supla/sensor/HX711.h:152]: This variable is not initialized

File: src/supla/sensor/HX711.h [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    bool _tare = false;
+    hx711Sensor.start(stabilizingTime, _tare);
+    if (hx711Sensor.getTareTimeoutFlag()) {
+      SUPLA_LOG_DEBUG("HX711 timeout, check connection");
+    } else {
+      calFactor = (calValue > 0) ? calValue : defCalValue;
+      hx711Sensor.setCalFactor(calFactor);
+      SUPLA_LOG_DEBUG("HX711 startup is complete");
+    }
+    while (!hx711Sensor.update()) {}
```
Comment (klew) [inline src/supla/sensor/HX711.h:69]: if update() will constantly return "false", then this line will hang whole program. Please add some timeout here.

File: src/supla/sensor/HX711.h [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    SUPLA_LOG_INFO("HX711 calibration value: %.2f", hx711Sensor.getCalFactor());
+    SUPLA_LOG_INFO(
+        "HX711 measured conversion time ms: %.2f",
+         hx711Sensor.getConversionTime());
+    SUPLA_LOG_DEBUG(
+        "HX711 measured sampling rate HZ: %.2f", hx711Sensor.getSPS());
+    SUPLA_LOG_INFO(
+        "HX711 measured settlingtime ms: %d", hx711Sensor.getSettlingTime());
```
Comment (klew) [inline src/supla/sensor/HX711.h:77]: settlingtime? missing space

File: src/supla/sensor/HX711.h [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    }
```
Comment (klew) [inline src/supla/sensor/HX711.h:80]: this check is not needed here. The same code will be executed in getValue which is called one line below

File: src/supla/sensor/HX711.h [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    tareOffset = hx711Sensor.getTareOffset();
+    SUPLA_LOG_DEBUG("HX711 new tare offset: %d", tareOffset);
+    Supla::Storage::ScheduleSave(100);
+    readyToCalibration = true;
+  }
+
+  void iterateAlways() {
+    if (hx711Sensor.update()) {
+      value = hx711Sensor.getData();
+    }
```
Comment (klew) [inline src/supla/sensor/HX711.h:112]: You can move  this code to below "if". Then it will call update and getData only every 500 ms, when it is needed.

**Review by klew — CHANGES_REQUESTED** [review]

File: src/supla/sensor/HX711.h (hunk 1 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h:149]: ok

File: src/supla/sensor/HX711.h (hunk 2 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: corrected

File: src/supla/sensor/HX711.h (hunk 3 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: corrected

File: src/supla/sensor/HX711.h (hunk 4 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: corrected

File: src/supla/sensor/HX711.h (hunk 5 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: corrected

File: src/supla/sensor/HX711.h (hunk 6 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: According to the library, the sampling frequency must be in the range 7 - 100 Hz. Now, it's around 10.5 Hz.

File: src/supla/sensor/HX711.h (hunk 1 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: Without these variables, compilation in the arduino IDE fails

Comment (klew) [reply src/supla/sensor/HX711.h:154]: Yes, you have to also remove them from initialization list of constructor

**Review by klew — APPROVED** [review]

File: src/supla/sensor/HX711.h (hunk 1 above)
Comment (lukfud) [reply src/supla/sensor/HX711.h]: Done

**Review by klew — APPROVED** [review]

**Comment by klew** [comment]
Thanks for your contribution!

**Pull request merged**